server/consumer.go

```diff
@@ -228,13 +230,13 @@ func (mset *Stream) AddConsumer(config *ConsumerConfig) (*Consumer, error) {
 
 func (mset *Stream) addConsumer(config *ConsumerConfig, oname string, node RaftNode) (*Consumer, error) {
 	mset.mu.RLock()
-	jsa := mset.jsa
+	s, jsa := mset.srv, mset.jsa
 	mset.mu.RUnlock()
 
 	// If we do not have the consumer assigned to us in cluster mode we can not proceed.
 	// Running in single server mode this always returns true.
 	if oname != _EMPTY_ && !jsa.consumerAssigned(mset.Name(), oname) {
-		return nil, ErrJetStreamNotAssigned
+		s.Debugf("Consumer %q > %q does not seem to be assigned to this server", mset.Name(), oname)
```

Comment:
So now we are not returning with an error like we used to. Is that intentional? The comment above seem to indicate that if not assigned we cannot proceed, yet we do.

Reply:
This is all for raciness during startup. We should warn but could be that we process a recovered stream or consumer before we process the assignment if it was the last. In normal practice does not happen and if it is truly not supposed to be there the deleteOp will follow from meta and delete it anyway.